src/Cafe/OS/libs/nsyshid/Dimensions.cpp

```diff
@@ -0,0 +1,1080 @@
+			else if (!m_queries.empty())
+			{
+				response = m_queries.front();
+				m_queries.pop();
+				responded = true;
+			}
+			else
+			{
+				std::this_thread::sleep_for(std::chrono::milliseconds(100));
+			}
+		}
+		while (responded == false);
+		return response;
+	}
+
+	void DimensionsUSB::SendCommand(std::span<const uint8, 32> buf)
+	{
+		const uint8 command = buf[2];
+		const uint8 sequence = buf[3];
+
+		std::array<uint8, 32> q_result{};
+
+		switch (command)
+		{
+		case 0xB0: // Wake
+		{
+			// Consistent device response to the wake command
+			q_result = {0x55, 0x0e, 0x01, 0x28, 0x63, 0x29,
+						0x20, 0x4c, 0x45, 0x47, 0x4f, 0x20,
+						0x32, 0x30, 0x31, 0x34, 0x46};
+			break;
+		}
+		case 0xB1: // Seed
+		{
+			// Initialise a random number generator using the seed provided
+			g_dimensionstoypad.GenerateRandomNumber(std::span<const uint8, 8>{buf.begin() + 4, 8}, sequence, q_result);
+			break;
+		}
+		case 0xB3: // Challenge
+		{
+			// Get the next number in the sequence based on the RNG from 0xB1 command
+			g_dimensionstoypad.GetChallengeResponse(std::span<const uint8, 8>{buf.begin() + 4, 8}, sequence, q_result);
+			break;
+		}
+		case 0xC0: // Color
+		case 0xC1: // Get Pad Color
+		case 0xC2: // Fade
+		case 0xC3: // Flash
+		case 0xC4: // Fade Random
+		case 0xC6: // Fade All
+		case 0xC7: // Flash All
+		case 0xC8: // Color All
+		{
+			// Send a blank response to acknowledge color has been sent to toypad
+			q_result = {0x55, 0x01, sequence};
+			q_result[3] = GenerateChecksum(q_result, 3);
+			break;
+		}
+		case 0xD2: // Read
+		{
+			// Read 4 pages from the figure at index (buf[4]), starting with page buf[5]
+			g_dimensionstoypad.QueryBlock(buf[4], buf[5], q_result, sequence);
+			break;
+		}
+		case 0xD3: // Write
+		{
+			// Write 4 bytes to page buf[5] to the figure at index buf[4]
+			g_dimensionstoypad.WriteBlock(buf[4], buf[5], std::span<const uint8, 4>{buf.begin() + 6, 4}, q_result, sequence);
+			break;
+		}
+		case 0xD4: // Model
+		{
+			// Get the model id of the figure at index buf[4]
+			g_dimensionstoypad.GetModel(std::span<const uint8, 8>{buf.begin() + 4, 8}, sequence, q_result);
+			break;
+		}
+		case 0xD0: // Tag List
+		case 0xE1: // PWD
+		case 0xE5: // Active
+		case 0xFF: // LEDS Query
+		{
+			// Further investigation required
+			cemuLog_log(LogType::Force, "Unimplemented LD Function: {:x}", command);
+			break;
+		}
+		default:
+		{
+			cemuLog_log(LogType::Force, "Unknown LD Function: {:x}", command);
+			break;
+		}
+		}
+
+		m_queries.push(q_result);
+	}
+
+	uint32 DimensionsUSB::LoadFigure(const std::array<uint8, 0x2D * 0x04>& buf, std::unique_ptr<FileStream> file, uint8 pad, uint8 index)
+	{
+		std::lock_guard lock(m_dimensionsMutex);
+		const uint32 id = GetFigureId(buf);
+
+		DimensionsMini& figure = GetFigureByIndex(index);
+		figure.dimFile = std::move(file);
+		figure.id = id;
+		figure.pad = pad;
+		figure.index = index + 1;
+		memcpy(figure.data.data(), buf.data(), buf.size());
```

Comment:
This memcpy can also be a simple assignment
```suggestion
		figure.data = buf;
```